google-cloudevent-types/src/main/java/com/google/events/cloud/audit/v1/AuditLog.java

```diff
@@ -0,0 +1,122 @@
+package com.google.events.cloud.audit.v1;
+
+import java.util.Map;
+
+/**
+ * The log entry payload, which is always an AuditLog for Cloud Audit Log events.
+ */
+public class AuditLog {
+    private AuthenticationInfo authenticationInfo;
+    private AuthorizationInfo[] authorizationInfo;
+    private Map<String, Object> metadata;
+    private String methodName;
+    private Long numResponseItems;
+    private Map<String, Object> request;
+    private RequestMetadata requestMetadata;
+    private ResourceLocation resourceLocation;
+    private String resourceName;
+    private Map<String, Object> resourceOriginalState;
+    private Map<String, Object> response;
+    private Map<String, Object> serviceData;
+    private String serviceName;
+    private Status status;
+
+    /**
+     * Authentication information.
+     */
+    public AuthenticationInfo getAuthenticationInfo() { return authenticationInfo; }
+    public void setAuthenticationInfo(AuthenticationInfo value) { this.authenticationInfo = value; }
+
+    /**
+     * Authorization information. If there are multiple resources or permissions involved, then
+     * there is one AuthorizationInfo element for each {resource, permission} tuple.
+     */
+    public AuthorizationInfo[] getAuthorizationInfo() { return authorizationInfo; }
+    public void setAuthorizationInfo(AuthorizationInfo[] value) { this.authorizationInfo = value; }
+
+    /**
+     * Other service-specific data about the request, response, and other information associated
+     * with the current audited event.
+     */
+    public Map<String, Object> getMetadata() { return metadata; }
+    public void setMetadata(Map<String, Object> value) { this.metadata = value; }
+
+    /**
+     * The name of the service method or operation. For example
+     * "google.datastore.v1.Datastore.RunQuery"
+     */
+    public String getMethodName() { return methodName; }
+    public void setMethodName(String value) { this.methodName = value; }
+
+    /**
+     * The number of items returned from a List or Query API method, if applicable.
+     */
+    public Long getNumResponseItems() { return numResponseItems; }
+    public void setNumResponseItems(Long value) { this.numResponseItems = value; }
+
+    /**
+     * The operation request. This may not include all request parameters, such as those that
+     * are too large, privacy-sensitive, or duplicated elsewhere in the log record. It should
+     * never include user-generated data, such as file contents. When the JSON object
+     * represented here has a proto equivalent, the proto name will be indicated in the `@type`
+     * property.
+     */
+    public Map<String, Object> getRequest() { return request; }
```

Comment:
Do we know what this does in terms of deserialization? (How does one get at sub-properties?)

Reply:
I think a dev would have to use their own object-to-pojo to get sub-properties. No data should be modified. I'm not sure what the proto does, but we could possibly expand the jsonschema definition here.